benchmarks/Microsoft.AspNetCore.Routing.Performance/Matchers/MatcherSelectCandidatesAzureBenchmark.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using BenchmarkDotNet.Attributes;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    // Generated from https://github.com/Azure/azure-rest-api-specs
+    public partial class MatcherSelectCandidatesAzureBenchmark : MatcherBenchmarkBase
+    {
+        private const int SampleCount = 100;
+
+        private BarebonesMatcher _baseline;
+        private DfaMatcher _dfa;
```

Comment:
This is the first of a set of 4 small lower level benchmarks for the DFA algorithm. I'm just testing the DFA here because the other ones don't have a similar layering.